src/ciphersuite/mod.rs

```diff
@@ -56,6 +56,24 @@ pub struct HpkeCiphertext {
     ciphertext: Vec<u8>,
 }
 
+/// A label which can be used to expand a `Secret` using an HKDF.
+struct HkdfLabel {
+    length: u16,
+    label: String,
+    context: Vec<u8>,
+}
+
+impl HkdfLabel {
+    pub fn new(context: &[u8], label: &str, length: usize) -> Self {
+        let full_label = "mls10 ".to_owned() + label;
```

Comment:
Created issue #231 for this. It involves pushing the `ProtocolVersion` all the way down to places where we derive keys, so I'd prefer not fixing this in this PR. I already just moved this part and didn't really touch it otherwise.